### ISSUE TITLE
Laravel Passport :The resource owner or authorization server denied the request

### ISSUE THREAD

**Issue by souravt1992** [laravel/passport#1093]
- Passport Version: "^7.2"
- Laravel Version:"5.8.0"
- PHP Version:"^7.1.3"
- Database Driver & Version:postgres

### Description:

we have used passport as api middle ware authentication in one our project and when user login and performs some action on the website after some time there is error which is coming on 
 production.ERROR: The resource owner or authorization server denied the request
by default token expiration time is 1 year but still there is error which we are not aware about

### Steps To Reproduce:
we have login for the same user from different machine means we have logeed it for user A from different Ip and then after performing some navgation  token expire error comes up for all machines.

**Comment by driesvints** [comment]
Hi there,

Thanks for reporting but it looks like this is a question which can be asked on a support channel. Please only use this issue tracker for reporting bugs with the library itself. If you have a question on how to use functionality provided by this repo you can try one of the following channels:

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)

However, this issue will not be locked and everyone is still free to discuss solutions to your problem!

Thanks.

**Comment by souravt1992** [comment]
While resolving this issue I have some learning so I want to share it across .

This issue is basically linked to Auth provider of Laravel i.e. while making a request from api to backend server in middle of some request there is EOF error.
For understanding this I have done debugging and gone through the documentation of Laravel and tried to interpret the scenario.
By default Laravel auth is design for web routes i.e. it works best with Laravel application frontend and backend as a single application but in our case we have used Laravel as REST  application for our application and we have done some modification in the auth.php and Kernel.php and api.php

So  now I will explain how its working . 
In auth.php we have this 

'defaults' => [
        'guard' => 'web',
        'passwords' => 'users',
    ],  

    'guards' => [
        'web' => [
            'driver' => 'session',
            'provider' => 'users',
        ],

        'api' => [
            'driver' => 'token',
            'provider' => 'users',
        ],
    ],

So according to this setting what was happening is by default web guards are working whenever we are calling AUTH::user() method to fetch user from authorization.
so whenever Auth guards are initiated they are linked to default web guards and web guard are linked to session driver but we have not created any linking  in our code to it as a result there is no session of user while using api call and there is EOF error in our production.

TO overcome this problem I have used    Auth::guard('api')->user(); this will direct our request to connect to correct guard while using Auth middleware.

we have also have the option to extend Auth service provider so I have implemented and extend in the following manner.

  Auth::viaRequest('token', function ($request) {
            $api_token=  $request->header('Authorization');
            $api_token=str_replace('Bearer ', '', $api_token);
           // $id=$request->header('id');
           // return User::where('api_token', $api_token)->where('id', $id)->first();
           return User::where('api_token', $api_token)->first();
       });   

After doing this I come to conclusion that there is no fault in passport or jwt-auth we have do some configuration changes in the right way and the need to direct those to the right end points on Laravel 😊
